gufe/tests/test_mapping.py

```diff
@@ -0,0 +1,64 @@
+# This code is part of gufe and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/gufe
+
+import gufe
+from gufe import AtomMapping
+import pytest
+
+
+from .test_tokenization import GufeTokenizableTestsMixin
+
+
+class ExampleMapping(AtomMapping):
+    def __init__(self, molA: gufe.SmallMoleculeComponent,
+                 molB: gufe.SmallMoleculeComponent, mapping):
+        self._molA = molA
+        self._molB = molB
+        self._mapping = mapping
+
+    def _defaults(self):
+        return {}
+
+    def _to_dict(self):
+        return {
+            'molA': self._molA,
+            'molB': self._molB,
+            'mapping': self._mapping,
+        }
+
+    @classmethod
+    def _from_dict(cls, d):
+        return cls(**d)
+
+    def molA_to_molB(self):
+        return self._mapping
+
+    def molB_to_molA(self):
+        return {v: k for k, v in self._mapping}
+
+    def molA_unique(self):
+        return (i for i in range(self._molA.to_rdkit().GetNumAtoms())
+                if i not in self._mapping)
+
+    def molB_unique(self):
+        return (i for i in range(self._molB.to_rdkit().GetNumAtoms())
+                if i not in self._mapping.values())
```

Comment:
Given their implementation isn't defined in the abstract class I don't really care too much here, but is there worth in testing these too for the sake of ensuring we don't introduce breaking changes without knowing?

Reply:
this should get tested in openfe for the actual implementations we use, these are just examples